collections/_article/whats-new-in-csharp-for-godot-4-0.md

```diff
@@ -0,0 +1,357 @@
+
+To help you choose which collection type to use and when, we have a new documentation page all about [_C# Collections_](https://docs.godotengine.org/en/latest/tutorials/scripting/c_sharp/c_sharp_collections.html).
+
+#### Relevant links
+
+- Merge .NET 6 branch with master ([GH-64089](https://github.com/godotengine/godot/pull/64089)).
+- C# Export on List not possible anymore ([GH-70298](https://github.com/godotengine/godot/issues/70298)).
+- Sync C# Array with Core ([GH-71786](https://github.com/godotengine/godot/pull/71786)).
+- Sync C# Dictionary with Core ([GH-71984](https://github.com/godotengine/godot/pull/71984)).
+
+## Signals as events
+
+Godot signals are a great way to implement the Observer pattern. In C#, it's possible to use Godot signals by calling the `Connect` and `EmitSignal` methods, but C# developers are more used to C# events as they feel more idiomatic.
+
+For Godot 4.0, we now generate C# events for all Godot signals, which allows developers to use the event syntax that they are used to and love. As a consequence, subscribing to Godot signals using the event syntax is type-safe.
+
+In contrast with normal C# events, these generated events differ from normal C# events in two ways:
+
+* These events are disconnected from automatically upon freeing the Godot type that contains the signal.
+* Using the `Invoke` method to raise the event is not allowed.
+
+We are looking into ways to allow raising events in a type-safe way, so keep an eye out for that in the future.
+
+For more information about how to use Signals in C#, take a look at the new documentation page about [_C# Signals_](https://docs.godotengine.org/en/latest/tutorials/scripting/c_sharp/c_sharp_signals.html).
+
+#### Relevant links
+
+- Fix C# bindings after recent breaking changes ([GH-37050](https://github.com/godotengine/godot/pull/37050)).
+- Create a dedicated "C# Signals" page ([GH-6643](https://github.com/godotengine/godot-docs/pull/6643)).
+
+## Callable support
+
+In Godot 4, we introduce the new `Callable` type that represents a method in an object instance or a standalone function, this may sound familiar because C# already supports a similar concept with delegate types, the `Action` and `Func` types, and lambdas.
+
+In order to support interoperability with the engine, Godot's C# API implements a `Callable` type that can be created from an `Action` or `Func`. This allows users to use C# lambdas with Godot API.
+
+`Callable`s can be invoked by using the `Call` or `CallDeferred` methods. We currently don't support binding values to the parameters in C# but this shouldn't be a problem when using C# lambdas since they can use closures to support this scenario.
```

Comment:
```suggestion
In Godot 4 we introduce the new `Callable` type that represents a method in an object instance or a standalone function. This may sound familiar, because C# already supports a similar concept with delegate types, the `Action` and `Func` types, and lambdas.

In order to support interoperability with the engine, Godot's C# API implements a `Callable` type that can be created from an `Action` or a `Func`. This also allows developers to use C# lambdas with Godot API.

`Callable`s can be invoked by using the `Call` or `CallDeferred` methods. We currently don't support binding values to the parameters in C#. This shouldn't be a problem when using C# lambdas since they can use closures to support this scenario.
```